src/app/hooks/useAcceptTerms.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from "react";
+import { useLocalStorage } from "usehooks-ts";
+
+import { postTerms } from "@/app/api/postTerms";
+import { useError } from "@/app/context/Error/ErrorContext";
+import { ErrorState } from "@/app/types/errors";
+
+export const useAcceptTerms = (address: string, publicKeyNoCoord: string) => {
```

Comment:
What is the logic that we are implementing here? A comment explaining what we aim to accomplish would be useful.